### PR TITLE
License Caching and Add license enforcement to all APIs

### PR DESCRIPTION
**Pull request by jashaik**
### Description

[Please describe what this change achieves]
Implementation of license cache and changes in web machine to add license enforcement to all APIs
https://progresssoftware.atlassian.net/browse/CHEF-15863
https://progresssoftware.atlassian.net/browse/CHEF-12649
### Issues Resolved

[List any existing issues this PR resolves, or any Discourse or
StackOverflow discussions that are relevant]

### Check List

- [ ] New functionality includes tests
- [ ] All buildkite tests pass
- [ ] Full omnibus build and tests in buildkite pass
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/main/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [ ] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

**Comment by jashaik** [comment]
Omnibus build - https://buildkite.com/chef/chef-chef-server-main-omnibus-adhoc/builds/6731

**Comment by jashaik** [comment]
https://buildkite.com/chef/chef-chef-server-main-habitat-build/builds/1246

**Comment by jashaik** [comment]
Manual testing results
Screenshot 2024-10-04 at 12 52 02 PM

**Comment by jashaik** [comment]
Screenshot 2024-10-04 at 12 57 21 PM

**Comment by jashaik** [comment]
Screenshot 2024-10-04 at 1 03 58 PM

File: src/oc_erchef/apps/oc_chef_wm/src/oc_chef_wm_base.erl [hunk 1]
```diff
@@ -84,7 +84,29 @@ service_available(Req, #base_state{reqid_header_name = HeaderName} = State) ->
             State1 = State0#base_state{server_api_version = Version},
             State2 = set_req_contexts(Req, State1),
             State3 = maybe_with_default_org(Req, State2),
-            {true, Req, State3}
+            case wrq:path(Req) of
+                "/_status" ->
+                    {true, Req, State3};
+                _ ->
+                    case chef_license_worker:get_license() of
```
Comment (marcparadise) [inline src/oc_erchef/apps/oc_chef_wm/src/oc_chef_wm_base.erl:91]: can we handle this case statement as a function, something like 
```
_ -> 
  check_license(chef_license_worker:get_license(), Req, State3) 
```

With: 
``` 
check_license({valid_license..... }, Req, State ) ->
  { true, Req, State} ;
check_license({commercial_grace_period...
... etc
```

Makes it a  little easier to test, and avoids deeply nesting the contents of `service_available` which makes it easier to parse.

**Comment by jashaik** [comment]
https://buildkite.com/chef/chef-chef-server-main-omnibus-adhoc/builds/6737

**Comment by jashaik** [comment]
Please find the below results: Working `x-ops-license` and _status API without license
Screenshot 2024-10-08 at 11 49 41 AM
Screenshot 2024-10-08 at 11 50 17 AM

**Review by lbakerchef — APPROVED** [review]

**Review by talktovikas — APPROVED** [review]
LGTM

**Pull request merged**